### PR TITLE
[FIXED JENKINS-35168] Resolve the context `ModelObject` when `it` isn't a `ModelObeject`

### PR DESCRIPTION
**Pull request by stephenc**
@reviewbybees

### CODE REVIEW

File: src/main/resources/lib/credentials/select.jelly [hunk 1]
```diff
@@ -71,8 +75,9 @@
                 default="${paramDefault?'':attrs.default}" value="${paramValue?(paramDefault?'':attrs.default):value}"/>
       <!-- TODO  add support for checking permissions against stores in request path -->
       <st:nbsp/>
+      <j:set var="storeItems" value="${selectHelper.getStoreItems(context, includeUser)}"/>
       <j:choose>
-        <j:when test="${h.hasPermission(selectHelper.CREATE)}">
+        <j:when test="${h.hasPermission(selectHelper.CREATE) and storeItems != null and !storeItems.isEmpty()}">
```
Comment (tfennelly) [inline src/main/resources/lib/credentials/select.jelly]: I didn't know `and` would work in those expressions :)

**Comment by tfennelly** [comment]
🐝

File: src/main/resources/lib/credentials/select.jelly (hunk 1 above)
Comment (oleg-nenashev) [inline src/main/resources/lib/credentials/select.jelly]: Seems to be another unrelated fix, but LGTM

**Comment by oleg-nenashev** [comment]
🐝

File: src/main/resources/lib/credentials/select.jelly (hunk 1 above)
Comment (stephenc) [inline src/main/resources/lib/credentials/select.jelly]: @tfennelly yes it is a special alias because typing `&amp;&amp;` would be a pain

Comment (stephenc) [inline src/main/resources/lib/credentials/select.jelly]: @oleg-nenashev not quite, if the resolution fails we will have an empty list so the button should be disabled

**Pull request merged**